pkg/operator/collection.go

```diff
@@ -364,7 +374,9 @@ type prometheusConfig struct {
 	SecretConfigs []secrets.SecretConfig `yaml:"kubernetes_secrets,omitempty"`
 }
 
-func (r *collectionReconciler) makeCollectorConfig(ctx context.Context, spec *monitoringv1.CollectionSpec, exports []monitoringv1.ExportSpec) (*prometheusConfig, error) {
+// makeCollectorConfig returns the Prometheus configuration, the list of objects to update, the list
+// of objects whose status need to be updated and any list of errors.
+func (r *collectionReconciler) makeCollectorConfig(ctx context.Context, spec *monitoringv1.CollectionSpec, exports []monitoringv1.ExportSpec) (*prometheusConfig, []monitoringv1.MonitoringCRD, []monitoringv1.MonitoringCRD, error) {
```

Comment:
suggestion: This may be a good place for named result parameters, since there are two parameters in a row with the same type.

go/go-style/decisions#named-result-parameters

Reply:
You can add `statusUpdates` and the new status back to the `collectionReconciler` struct.

Reply:
I now return a `[]update` instead.